### DISCUSSION TITLE
Top Languages Card having issues with picture and source elements for light and dark mode

### DISCUSSION THREAD

**Discussion by kennethjmyers** [anuraghazra/github-readme-stats#3723] [Q&A] [answered]
### Describe the bug

I'm trying to use the `picture` element to switch between light and dark mode views for the top-langs card but I am running into visual issues that I don't see with the stats card. I'm including two tests, one for the version I'm trying to include in my readme, and one where I remove the light mode `source`. I'm noticing a few things:

1. Light mode and dark mode get used, but everything after the first query seems to get ignored. 
    1. in my first test HTML is placed first in the list of languages to hide. This indeed seems to get filtered out, but we can see jupyter notebook and css get left in and are not filtered out as expected.
    2. In my second test, jupyter notebook is placed first. Now jupyter notebook is filtered out but HTML is allowed in
 2. This seems mainly an issue with the source element because if you remove the light-mode source and let it default to the `img` tag, then light mode turns out fine (third test)

What's odd is I'm only seeing this issue on github and not my IDE's preview (I've included screenshots of this, I'm using pycharm in case that's relevant). So Is this somehow an issue on Github's end? I have tested this in chrome and safari and I'm seeing the same results.

## First test (copied from readme example, but changed the src link so it uses the top-langs card):

<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook" />
</picture>

html:

```html
<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook" />
</picture>
```

What I'm seeing right now on github (dark mode):
![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/421dcf4f-2c17-4b31-8e6e-5b194f2036fc)

What I see in my IDE preview (dark mode):

![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/613213a3-8b2d-4d43-85b4-ce9fc4e93a1e)

IDE preview looks right.

What I'm seeing on github (light mode):

![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/4890d3c8-e607-477a-8e4a-b09ccaca0aa6)

Both light mode and dark mode filter out HTML, but not the others

## Second Test - put jupyter notebook first

<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=Jupyter%20Notebook,HTML,CSS,SCSS,Ruby"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=Jupyter%20Notebook,HTML,CSS,SCSS,Ruby"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=Jupyter%20Notebook,HTML,CSS,SCSS,Ruby" />
</picture>

html:

```html
<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=Jupyter%20Notebook,HTML,CSS,SCSS,Ruby"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=Jupyter%20Notebook,HTML,CSS,SCSS,Ruby"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=Jupyter%20Notebook,HTML,CSS,SCSS,Ruby" />
</picture>
```

What I'm seeing right now on github (dark mode):
![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/415d169d-fea9-46d4-ad9a-8e746e676fe8)

Jupyter notebook is filtered out (because it is first) but HTML is not filtered out.

What I see in my IDE preview (dark mode):

![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/613213a3-8b2d-4d43-85b4-ce9fc4e93a1e)

IDE preview still looks right

## Third Test - remove the source element for light color-scheme

<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook"
    media="(prefers-color-scheme: dark)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook" />
</picture>

html:

```html
<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook"
    media="(prefers-color-scheme: dark)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML,CSS,SCSS,Ruby,Jupyter%20Notebook" />
</picture>
```

What I'm seeing right now on github (dark mode):

![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/78197991-8f01-4b86-ab69-1a660bb60cbd)

What I see in my IDE preview (dark mode):

(same as first test, looks correct)

What I'm seeing on github (light mode):

![image](https://github.com/anuraghazra/github-readme-stats/assets/13514551/b31b2abe-34c3-4be2-9e2d-8c6c51ccdc6c)

So it seems correct if you don't try to use the `source` element for light mode.

### Expected behavior

All of the query elements should be filtered out, not just the first onw. 

### Screenshots / Live demo link

see above

**Comment by qwerty541** [comment, marked as answer]
Hey, @kennethjmyers! You have to URL encode the content of `hide` query string parameter when using `picture` and `source` elements. I will add an example below.

```html
<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=HTML%2CCSS%2CSCSS%2CRuby%2CJupyter%20Notebook"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML%2CCSS%2CSCSS%2CRuby%2CJupyter%20Notebook"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML%2CCSS%2CSCSS%2CRuby%2CJupyter%20Notebook" />
</picture>
```

<picture>
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&theme=dark&hide=HTML%2CCSS%2CSCSS%2CRuby%2CJupyter%20Notebook"
    media="(prefers-color-scheme: dark)"
  />
  <source
    srcset="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML%2CCSS%2CSCSS%2CRuby%2CJupyter%20Notebook"
    media="(prefers-color-scheme: light), (prefers-color-scheme: no-preference)"
  />
  <img src="https://github-readme-stats.vercel.app/api/top-langs?username=kennethjmyers&show_icons=true&locale=en&layout=compact&hide=HTML%2CCSS%2CSCSS%2CRuby%2CJupyter%20Notebook" />
</picture>

**Reply by kennethjmyers** [reply]
Thank you so much! web stuff is not really something I'm great at and I had a feeling there was something like this I must be missing.

**Reply by qwerty541** [reply]
Glad to help! Feel free to contact me if you have any more questions.
